precode.go

```diff
@@ -1,55 +1,111 @@
 package main
 
 import (
-	"fmt"
-	"net/http"
-
-	"github.com/go-chi/chi/v5"
+    "encoding/json"
+    "fmt"
+    "log"
+    "net/http"
+    "strconv"
+    "github.com/go-chi/chi/v5"
 )
 
 // Task ...
 type Task struct {
-	ID           string   `json:"id"`
-	Description  string   `json:"description"`
-	Note         string   `json:"note"`
-	Applications []string `json:"applications"`
+    ID           string   `json:"id"`
+    Description  string   `json:"description"`
+    Note         string   `json:"note"`
+    Applications []string `json:"applications"`
 }
 
 var tasks = map[string]Task{
-	"1": {
-		ID:          "1",
-		Description: "Сделать финальное задание темы REST API",
-		Note:        "Если сегодня сделаю, то завтра будет свободный день. Ура!",
-		Applications: []string{
-			"VS Code",
-			"Terminal",
-			"git",
-		},
-	},
-	"2": {
-		ID:          "2",
-		Description: "Протестировать финальное задание с помощью Postmen",
-		Note:        "Лучше это делать в процессе разработки, каждый раз, когда запускаешь сервер и проверяешь хендлер",
-		Applications: []string{
-			"VS Code",
-			"Terminal",
-			"git",
-			"Postman",
-		},
-	},
+    "1": {
+        ID:          "1",
+        Description: "Сделать финальное задание темы REST API",
+        Note:        "Если сегодня сделаю, то завтра будет свободный день. Ура!",
+        Applications: []string{
+            "VS Code",
+            "Terminal",
+            "git",
+        },
+    },
+    "2": {
+        ID:          "2",
+        Description: "Протестировать финальное задание с помощью Postmen",
+        Note:        "Лучше это делать в процессе разработки, каждый раз, когда запускаешь сервер и проверяешь хендлер",
+        Applications: []string{
+            "VS Code",
+            "Terminal",
+            "git",
+            "Postman",
+        },
+    },
 }
 
-// Ниже напишите обработчики для каждого эндпоинта
-// ...
+// getAllTasksHandler возвращает все задачи
+func getAllTasksHandler(w http.ResponseWriter, r *http.Request) {
+    w.Header().Set("Content-Type", "application/json")
 
-func main() {
-	r := chi.NewRouter()
+    allTasks := make([]Task, 0, len(tasks))
+    for _, task := range tasks {
+        allTasks = append(allTasks, task)
+    }
+
+    json.NewEncoder(w).Encode(allTasks)
```

Comment:
А зачем просто создавать копию массива? Можно просто его сериализовать
```suggestion
    json.NewEncoder(w).Encode(tasks)
```